### ISSUE TITLE
BatchNormalization is broken for validation

### ISSUE THREAD

**Issue by ielenik** [ROCmSoftwarePlatform/tensorflow-upstream#1289]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

### System information 
Ubuntu 20.04 lts
Gfx906

-   **Have I written custom code (as opposed to using a stock example script
    provided in TensorFlow)**: YES
-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 20.04
-   **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue
    happens on a mobile device**: No
-   **TensorFlow installed from (source or binary)**: binary
-   **TensorFlow version (use command below)**: 2.4.0
-   **Python version**: 3.8
-   **GPU model and memory**: Gfx906
-   **Exact command to reproduce**: run provided python file

### Describe the problem
There is a problem with simple training validation in case BatchNormalization layers exists for 4d tensor

### Source code / logs

`
import tensorflow as tf

from tensorflow.keras.layers import Dense, Dropout, Activation, Flatten, Conv2D
from tensorflow.keras.layers import Input, MaxPooling2D, Dropout, BatchNormalization
from tensorflow.keras.models import Model
import numpy as np

x = np.random.uniform(-1,1,(10000,4,4,1))
y = x.reshape(10000,16)

#*************************************************
print('*'*20)

o = inp = Input(shape=(4,4,1))
o = BatchNormalization()(o)
o = Flatten()(o)

m = Model(inp,o)
m.compile(loss='MSE',optimizer='SGD')

m.fit(x,y,batch_size=16,validation_split=0.01,epochs=5,verbose=1)

#*************************************************
print('*'*20)

o = inp = Input(shape=(4,4,1))
o = Flatten()(o)

m = Model(inp,o)
m.compile(loss='MSE',optimizer='SGD')

m.fit(x,y,batch_size=16,validation_split=0.01,epochs=5,verbose=1)

#*************************************************
print('*'*20)

o = inp = Input(shape=(4,4))
o = BatchNormalization()(o)
o = Flatten()(o)

m = Model(inp,o)
m.compile(loss='MSE',optimizer='SGD')

m.fit(x,y,batch_size=16,validation_split=0.01,epochs=5,verbose=1)
`
The output:

`: Vega 20 [Radeon VII], pci bus id: 0000:08:00.0)
2021-03-15 23:48:43.581643: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:116] None of the MLIR optimization passes are enabled (registered 2)
2021-03-15 23:48:43.582177: I tensorflow/core/platform/profile_utils/cpu_utils.cc:112] CPU Frequency: 2993990000 Hz
Epoch 1/5
2021-03-15 23:48:43.805724: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libMIOpen.so
619/619 [==============================] - 2s 2ms/step - loss: 0.0294 - val_loss: 18.5235
Epoch 2/5
619/619 [==============================] - 1s 2ms/step - loss: 0.0015 - val_loss: 19.8782
Epoch 3/5
619/619 [==============================] - 1s 2ms/step - loss: 0.0014 - val_loss: 20.7306
Epoch 4/5
619/619 [==============================] - 1s 2ms/step - loss: 0.0014 - val_loss: 20.8199
Epoch 5/5
619/619 [==============================] - 1s 2ms/step - loss: 0.0015 - val_loss: 20.8838
********************
Epoch 1/5
619/619 [==============================] - 1s 1ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 2/5
619/619 [==============================] - 1s 1ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 3/5
619/619 [==============================] - 1s 1ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 4/5
619/619 [==============================] - 1s 1ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 5/5
619/619 [==============================] - 1s 1ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
********************
Epoch 1/5
619/619 [==============================] - 1s 2ms/step - loss: 0.0767 - val_loss: 3.3559e-04
Epoch 2/5
619/619 [==============================] - 1s 2ms/step - loss: 0.0065 - val_loss: 4.2532e-06
Epoch 3/5
619/619 [==============================] - 1s 2ms/step - loss: 0.0062 - val_loss: 5.8866e-06
Epoch 4/5
619/619 [==============================] - 1s 2ms/step - loss: 0.0062 - val_loss: 5.2724e-06
Epoch 5/5
619/619 [==============================] - 1s 2ms/step - loss: 0.0063 - val_loss: 5.2393e-06
`

Expected behavior:
Val_loss is acceptable

In case of 4d tensor and existed batch normalization layer validation loss is incorrect

That is output from first example on cuda
`Epoch 1/5                                                                                                                                         2021-03-15 23:52:36.620596: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cudnn64_8.dll     619/619 [==============================] - 2s 2ms/step - loss: 0.0288 - val_loss: 1.7430e-05                                                      Epoch 2/5                                                                                                                                         619/619 [==============================] - 1s 2ms/step - loss: 0.0017 - val_loss: 4.7978e-06                                                      Epoch 3/5                                                                                                                                         619/619 [==============================] - 1s 2ms/step - loss: 0.0017 - val_loss: 3.6881e-06                                                      Epoch 4/5                                                                                                                                         619/619 [==============================] - 1s 2ms/step - loss: 0.0015 - val_loss: 1.2209e-05                                                      Epoch 5/5                                                                                                                                         619/619 [==============================] - 1s 2ms/step - loss: 0.0016 - val_loss: 4.8993e-06 `

Looks like regression was after TF2.1

**Comment by sunway513** [comment]
@ekuznetsov139 can you help review this issue?

**Comment by ielenik** [comment]
Is not reproduced on 4.3
